scripts/app.js

```diff
@@ -12,16 +11,42 @@ function blog (data) {
   this.category = data.category;
 }
 
+blog.all = [];
+
 blog.prototype.toHtml = function() {
   var template = Handlebars.compile($('#article_template').text());
   this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
   return template(this);
 };
 
-blogData.forEach(function(ele) {
-  blogs.push(new blog(ele));
-});
+blog.fetchAll = function () {
+  if (localStorage.blogData) {
+    $.ajax({
+      type: 'HEAD',
+      url: 'data/blogData.json',
+      success: function(data,message,xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          localStorage.eTag = eTag;
```

Comment:
After you update the eTag here, you should also be updating the data in local storage.